src/v2/ion/responseTransformer.js

```diff
@@ -23,108 +23,11 @@ import { _ } from 'okta';
 
 /**
  * @typedef {Object} AuthnResult
- * @property {CurrentState} currentState
  * @property {Object=} user
  * @property {Object=} factor
  */
 
-/**
- * @typedef {Object} CurrentState
- * @property {string} stateHandle
- * @property {string} step
- * @property {string} intent
- * @property {function} cancel
- * @property {function} context
- * @property {Remediation} remediation
- */
-
-/**
- * @typedef {Object} Remediation
- * @property {RemediationObject[]} values
- */
-
-/**
- * @typedef {Object} RemediationObject
- * @property {string} name
- * @property {value[]} value mandatory visible parameters
- * @property {uiSchema[]} uiSchema `value` corresponding UI Schema
- */
-
 const isObject = x => _.isObject(x);
-const isArray = x => Array.isArray(x);
-const containsAny = (xs, ys) => _.intersection(xs, ys).length > 0;
-
-const getRelObjectByName = (resp, relName) => {
-  const getRelBy_ = (result_, resp_) => {
-    if (isObject(resp_)) {
-      if (resp_.rel && containsAny(resp_.rel, relName)) {
-        result_.push(resp_);
-      } else {
-        _.mapObject(resp_, getRelBy_.bind({}, result_));
-      }
-    } else if (isArray(resp_)) {
-      resp_.forEach(getRelBy_.bind({}, result_));
-    }
-    return result_;
-  };
-
-  return getRelBy_([], resp);
-};
-
-/**
- *
- * @returns {Object} actionObject
- * @returns {string} actionObject.name
- * @returns {function} actionObject.createRequest
- */
-const createActionObj = (relObj) => {
-  const data = (relObj.value || [])
-    .filter(v => v.visible === false)
-    .map(v => { return { [v.name]: v.value }; })
-    .reduce((init, x) => Object.assign({}, init, x), {});
-
-  const needExtraData = relObj.value.length !== Object.keys(data).length;
-  const method = relObj.method;
-  const url = relObj.href;
-  const obj = {
-    name: relObj.name,
-    createRequest (extraData = {}) {
-      const reqData = needExtraData ? Object.assign({}, data, extraData) : data;
-      return {
-        method,
-        url,
-        data: reqData,
-      };
-    }
-  };
-  if (relObj.refresh) {
-    obj.createRequest.refresh = relObj.refresh;
-  }
-
-  return obj;
-};
-
-/**
- * @returns {Object.<string, function>} actions
- */
-const createActions = (allCreateForms) => {
-  const allActions = allCreateForms.map(createActionObj);
-
-  // turns list of actions into a mapping
-  return allActions.reduce((init, action) => {
-    return Object.assign({}, init, {
-      [action.name]: action.createRequest,
-    });
-  }, {});
-};
-
-const convertRelFormToFunction = (value) => {
-  const allCreateForms = getRelObjectByName(value, ['create-form']);
-  const actions = createActions(allCreateForms);
-  const restValue = _.omit.apply(_, [value].concat(Object.keys(actions)));
-
-  return Object.assign({}, restValue, actions);
-};
 
 /**
  * Transform 'type: "object"' by converting "rel: form" to be a function.
```

Comment:
please update comment. Also add a sample input and output obj . 

eg) 

 {
  proceed: fn., 
  neededToProceed: {..},
  ....
}

to 

 
  neededToProceed: {..},
  ....
}

It is worth mentioning/capturing what we are retaining here with an example.